nomad/resource_dynamic_host_volume_registration_test.go

```diff
@@ -0,0 +1,211 @@
+		PreCheck:  func() { testAccPreCheck(t); testCheckMinVersion(t, minVersionDHV) },
+		Steps: []resource.TestStep{
+			{
+				Config: testResourceDynamicHostVolumeRegistration_config(name, nodeID),
+				Check: testResourceDynamicHostVolumeRegistration_check(
+					testResourceNameDynamicHostVolumeRegister, name),
+			},
+			{
+				ResourceName:      testResourceNameDynamicHostVolumeRegister,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+		CheckDestroy: testResourceDynamicHostVolume_checkDestroy(
+			testResourceNameDynamicHostVolumeRegister),
+	})
+}
+
+func TestResourceDynamicHostVolumeRegistration_refresh(t *testing.T) {
+	name := acctest.RandomWithPrefix("tf-nomad-test")
+	nodeID := testResourceDynamicHostVolume_getNodeID(t)
+
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testAccPreCheck(t); testCheckMinVersion(t, minVersionDHV) },
+		Steps: []resource.TestStep{
+			{
+				Config: testResourceDynamicHostVolumeRegistration_config(name, nodeID),
+				Check: testResourceDynamicHostVolumeRegistration_check(
+					testResourceNameDynamicHostVolumeRegister, name),
+			},
+			{
+				// causes volume to be deleted, exercising the "exists" function
+				PreConfig: testResourceDynamicHostVolume_delete(t, name),
+				Config:    testResourceDynamicHostVolumeRegistration_config(name, nodeID),
+			},
+		},
+		CheckDestroy: testResourceDynamicHostVolume_checkDestroy(
+			testResourceNameDynamicHostVolumeRegister),
+	})
+}
+
+func TestResourceDynamicHostVolumeRegistration_update(t *testing.T) {
+	name := acctest.RandomWithPrefix("tf-nomad-test")
+	nodeID := testResourceDynamicHostVolume_getNodeID(t)
+
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testAccPreCheck(t); testCheckMinVersion(t, minVersionDHV) },
+		Steps: []resource.TestStep{
+			{
+				Config: testResourceDynamicHostVolumeRegistration_config(name, nodeID),
+				Check: testResourceDynamicHostVolumeRegistration_check(
+					testResourceNameDynamicHostVolumeRegister, name),
+			},
+			{
+				Config: testResourceDynamicHostVolumeRegistration_update(name, nodeID),
+				Check: testResourceDynamicHostVolumeRegistration_update_check(
+					testResourceNameDynamicHostVolumeRegister, name),
+			},
+		},
+		CheckDestroy: testResourceDynamicHostVolume_checkDestroy(
+			testResourceNameDynamicHostVolumeRegister),
+	})
+}
+
+func testResourceDynamicHostVolumeRegistration_config(name, nodeID string) string {
+	return fmt.Sprintf(`
+resource "nomad_dynamic_host_volume_registration" "test" {
+  name      = "%s"
+  capacity  = "12 GiB"
+  node_id   = "%s"
+  host_path = "/tmp" # not really a good path but lets tests run
+
+  capability {
+    access_mode     = "single-node-writer"
+    attachment_mode = "file-system"
+  }
+
+  capability {
+    access_mode     = "single-node-reader-only"
+    attachment_mode = "file-system"
+  }
+
+  parameters = {
+    some_key = "some_value"
+  }
```

Comment:
tangential to TF, but isn't it weird to have `parameters` in a registration? that's entirely for plugins, isn't it? and registration doesn't use plugins.

Reply:
The API supports it but you're right, it just becomes silent metadata in the register workflow. We could remove it from the API by stripping it out in the Register RPC, but the object is always going to have the fields available. Maybe we should drop it from the TF provider? We should probably update the [parameters](https://developer.hashicorp.com/nomad/docs/v1.10.x/other-specifications/volume/host#parameters) docs to say those only apply to the create workflow as well (although it's hinted at by talking about plugins).

Reply:
On a second pass, I think we'll want to keep them in the provider, because the data source could be pulling from a volume that was created with either a register or create workflow and we can't know which. We could probably have a separate read-back method for the register resource, but then we'd end up with a ton of duplicated code to drift or a weird "don't include this one" parameter on the read path. What do you think?

Also, ironically, when running these against the rc.1 build, I had to remove the `parameter` block from our test configs for the create workflow, because that's the `mkdir` plugin which will now reject those.

Reply:
you can keep em in there if you want, so long as they're valid. I switched `foo = "bar"`-type stuff to `mode = "0755"` in some tests in `nomad` repo.

Reply:
:facepalm: of course! Fixed.